### PR TITLE
Minor enhancements for redirected links

### LINKED ISSUE #9: Borken google.com/about
When I try to download <https://google.com/about>, I get a broken page (missing links, broken formatting, etc.). I'm not sure what's missing, but as a comparison point: when I download the same page with [monolith](https://github.com/Y2Z/monolith), I get a 56MB file vs 53MB with obelisk.

### PR DESCRIPTION
**Pull request by waybackarchiver**
Closes #9

### CODE REVIEW

**Review by fmartingr — APPROVED** [review]
This seems good! Good touch adding the requested URL to the final archive :D

My only "concern" (if you can call it that) is that the first time I read "Origin" I was thinking about the header, not something related to Obelisk. But honestly, I can't think of a better name for this feature.

**Comment by waybackarchiver** [comment]
> Will this work if there are multiple reditects?

Multiple redirects do not work, my first sense is that most scenarios where redirects occur are usually only once, but now that you mention it, I do feel some need to deal with this scenarios.

**Comment by fmartingr** [comment]
> Multiple redirects do not work, my first sense is that most scenarios where redirects occur are usually only once, but now that you mention it, I do feel some need to deal with this scenarios.

Yeah, most redirects should be `A -> B`, not sure if we should consider covering more than that at the moment. Mostly because we need to handle how many redirect are we going to follow (in case of a broken link with infinite redirects or something like that). Or maybe the head method already follows all, never thought of that.

Anyway, if this is not a concern at the moment, it can be left as an issue until it is.

**Comment by waybackarchiver** [comment]
It should now be able to handle multiple redirects at https://github.com/go-shiori/obelisk/pull/11/commits/f77c37e4770574f5a458185e0fad5dc1eb67c8a3

File: archiver.go [hunk 1]
```diff
@@ -139,8 +145,39 @@ func (arc *Archiver) Archive(ctx context.Context, req Request) ([]byte, string,
...
+		if err != nil {
+			return u
+		}
+
+		if !isRedirect(resp) {
+			return resp.Request.URL
+		}
+		u = resp.Request.URL
+	}
+
```
Comment (fmartingr) [inline archiver.go:175]: We may need to log a warning/error here if `max == 0`.

**Comment by fmartingr** [comment]
> It should now be able to handle multiple redirects at [f77c37e](https://github.com/go-shiori/obelisk/commit/f77c37e4770574f5a458185e0fad5dc1eb67c8a3)

Awesome work! :D

File: archiver.go (hunk 1 above)
Comment (waybackarchiver) [reply archiver.go:182]: Done, when the maximum number of redirects (including other errors) is reached, an error is returned and no further action is taken.

**Comment by waybackarchiver** [comment]
Reviewed again the documentation about `http.Head`, in fact head requests have been implemented to follows multiple redirects automatically, perhaps we are redundant here?

```shell
% go doc http.Head
package http // import "net/http"

func Head(url string) (resp *Response, err error)
    Head issues a HEAD to the specified URL. If the response is one of the
    following redirect codes, Head follows the redirect, up to a maximum of 10
    redirects:

        301 (Moved Permanently)
        302 (Found)
        303 (See Other)
        307 (Temporary Redirect)
        308 (Permanent Redirect)
```

**Comment by fmartingr** [comment]
Yeah. It seems that it does exactly what you implemented. Sorry for not checking that before asking. 😥

**Comment by waybackarchiver** [comment]
> Yeah. It seems that it does exactly what you implemented. Sorry for not checking that before asking.

It's all right. I'll easily revert both commits https://github.com/go-shiori/obelisk/pull/11/commits/309437c0905d431778069378bf5a447845abf072 and https://github.com/go-shiori/obelisk/pull/11/commits/309437c0905d431778069378bf5a447845abf072.

**Comment by waybackarchiver** [comment]
We'll merge this improvement now and adjust it later if any other issues arise.

**Pull request merged**